include/nuttx/wiegand/wiegand.h

```diff
@@ -0,0 +1,88 @@
+/****************************************************************************
+ * include/nuttx/wiegand/wiegand.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_WIEGAND_WIEGAND_H
+#define __INCLUDE_NUTTX_WIEGAND_WIEGAND_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/irq.h>
+struct wiegand_gpio_s
+{
+    uint32_t gpio;
+    uint32_t id;
+};
+
+enum wiegand_status_e
+{
+  /* Parity sent was wrong. */
+
+  WIEGAND_PARITY_ERROR = -2,
+
+  /* The data read was uncompleted. */
+
+  WIEGAND_READ_UNCOMPLETED = -1,
+
+  /* The data read successfully. */
+
+  WIEGAND_SUCCESS = 0
+};
+
+struct wiegand_data_s
+{
+    uint32_t id;
+    int aba_code;
+    uint32_t facility_code;
+    enum wiegand_status_e status;
+};
+
+struct wiegand_config_s
```

Comment:
should we decouple the concrete driver implementation from the framework?
https://cwiki.apache.org/confluence/display/NUTTX/Upper+Half+and+Lower+Half+Drivers